### ISSUE TITLE
White screen not going away

### ISSUE THREAD

**Issue by astrovsky01** [romanhaa/Cerebro#14]
Even when reloading, the white screen is remaining, and i can't access the app past it. My hardware isn't particularly slow, so I don't know why.

**Comment by romanhaa** [comment]
Hi Alex. Sorry for the late reply. Remotely it's a bit hard to say what's going wrong on your end but this might be valuable feedback since I can't test Cerebro on all possible hardware setups.

Are you using macOS or Windows? If you are on macOS, there should be a log file in `~/Library/Logs/Cerebro/`. On Windows it should be in `C:\Users\<user>\AppData\Roaming\Cerebro`.

Best,
Roman

**Comment by astrovsky01** [comment]
No problem Roman. I'm working on integrating Cerebro as an interactive environment into Galaxy (https://usegalaxy.org/), so let me know whatever feedback you need.

I'm running on macOS Mojave 10.14.5, but while I was exploring code, I found that it runs on port 8080, which I was using for Galaxy at the same time. Loading it after exiting Galaxy, everything runs fine.

While I have your attention, I wanted to let you know that I'm making a conda package for your CerebroPrepare tool.

**Comment by romanhaa** [comment]
Good to hear you were able to solve it! And also interesting plan to integrate Cerebro into Galaxy. I'm curious to see how the performance will be but this could be very useful for a lot of people! 👌

Let me know if you need any help somewhere/somehow.

Best,
Roman

**Comment by astrovsky01** [comment]
Where is biocmanager pulling the tool prepare from? Most of the tools which use it are through bioconductor, but cerebro doesn't have a page for it.

**Comment by romanhaa** [comment]
cerebroPrepare is installed from its respective GitHub repo: `romanhaa/cerebroPrepare`. The `BiocManager::install()` function automatically installs from GitHub if the package name contains a `/`. Is that what you mean?

**Comment by astrovsky01** [comment]
Roman, that was what I meant, yes. Thanks!

**Comment by astrovsky01** [comment]
The initial screen for the app requests specifically an rds file, but doesn't mention a crb file, by the way.

**Comment by romanhaa** [comment]
Good catch :) I already included it in the new changes that will come. Not sure when I have the time to finalize it, but it seems to be working well. Brief overview:
* New function in cerebroPrepare that allows to extract a trajectory from a Monocle object and add it to the Seurat object which will then be exported for Cerebro.
* New panel in Cerebro that allows to explore the extracted trajectories (yes they can be multiple).
* Option to switch off WebGL for interactive plots which has major performance drawbacks but it seems that some peoples computers have problems with it and I don't understand how to fix/control that yet. For those who have problems with it, the dimensional reductions just stay empty. Scales are drawn correctly but there aren't any cells visible. This should help in those cases.

**Comment by astrovsky01** [comment]
Oh cool! I'm just finishing writing a workflow for cerebroPrepare from tsv to crb for a plug-and-play in galaxy, so once the new version is out, I'll add in the new functionality.

**Comment by astrovsky01** [comment]
Do you want to be a maintainer on the conda package? It's ready to upload to bioconda.

**Comment by romanhaa** [comment]
Could you take a screenshot of that? I don't quite understand what you mean. I'm guessing that the same issues happens in the example page of plotly (which is the framework I use in Cerebro)? You can find the example here: https://plot.ly/r/3d-scatter-plots/

**Comment by astrovsky01** [comment]
![labels_cover_scale](https://user-images.githubusercontent.com/40246333/59716477-0f7e0980-91e4-11e9-9113-bb8d39495d79.png)
![labels_cover_scale2](https://user-images.githubusercontent.com/40246333/59716478-0f7e0980-91e4-11e9-8354-82372fdab785.png)

From what I can tell from plotly, it's because the axis labels perpendicular to the axes themselves

**Comment by romanhaa** [comment]
Oh, I see. That's indeed not idea. I'll have a look at it.

**Comment by romanhaa** [comment]
I think this might be something changed in a recent plotly update since I haven't made any changes to the code and now it looks like this: 
![newplot](https://user-images.githubusercontent.com/18103560/59741477-6549c480-926b-11e9-8616-7258a31efa98.png)
